aiida/manage/configuration/migrations/migrations.py

```diff
@@ -344,6 +344,36 @@ def downgrade(self, config: ConfigType) -> None:
             profiles[profile_name] = profile
 
 
+class AddPrefixToStorageBackendTypes(SingleMigration):
+    """The ``storage.backend`` key should be prefixed with ``core.``.
+
+    At this point, it should only ever contain ``psql_dos`` which should therefore become ``core.psql_dos``. To cover
+    for cases where people manually added a read only ``sqlite_zip`` profile, we also migrate that.
+    """
+    down_revision = 8
+    down_compatible = 8
+    up_revision = 9
+    up_compatible = 9
+
+    def upgrade(self, config: ConfigType) -> None:
+        for profile_name, profile in config.get('profiles', {}).items():
+            if 'storage' in profile:
+                storage = profile['storage']
+                if 'backend' in storage:
+                    if storage['backend'] in ('psql_dos', 'sqlite_zip'):
+                        storage['backend'] = 'core.' + storage['backend']
+                    else:
+                        CONFIG_LOGGER.warning(
+                            f'profile {profile_name!r} had unknown storage backend {storage["backend"]!r}'
+                        )
+
+    def downgrade(self, config: ConfigType) -> None:
+        for profile in config.get('profiles', {}).values():
+            backend = profile.get('storage', {}).get('backend', None)
+            if backend is not None and backend.startswith('core.'):
+                profile.setdefault('storage', {})['backend'] = backend[5:]
```

Comment:
Mmm I'm not sure this is the way we want to do the downgrade. I'm thinking of the case that I have a `v9-config` with `storage['backend'] = core.new_backend`, and I downgrade to a version with `v8-config` to check something and come back. The downgrade will take away the `core`, but when upgrading it will not recognize the `new_backend` and just leave it as `storage['backend'] = new_backend`, which is now broken.

The generic recognition you are doing is good in general, but I think here we need to specifically only take out the `core` from 'psql_dos' and 'sqlite_zip' like in the upgrade.

It would also be nice to have a warning here too if the profile is neither of those two saying something like "downgrading to an older version of config where storage backends were not pluginable, your current storage won't work here" or something like that.

Reply:
Fair, I have made the downgrade only explicit for the known storage backends at this point. I have also added `sqlite_temp` which was missing.